backend/handlers/bets/betutils/feeutils_test.go

```diff
@@ -0,0 +1,198 @@
+					InitialBetFee int64 `yaml:"initialBetFee"`
+					BuySharesFee  int64 `yaml:"buySharesFee"`
+					SellSharesFee int64 `yaml:"sellSharesFee"`
+				}{
+					InitialBetFee: 1,
+					BuySharesFee:  0,
+					SellSharesFee: 0,
+				},
+			},
+		},
+	}
+}
+
+func TestGetUserInitialBetFee(t *testing.T) {
+	// Set up in-memory SQLite database
+	db, err := gorm.Open(sqlite.Open(":memory:"), &gorm.Config{})
+	if err != nil {
+		t.Fatalf("Failed to connect to the database: %v", err)
+	}
+
+	// Migrate the Bet and User models
+	db.AutoMigrate(&models.Bet{}, &models.User{})
+
+	// Mock the appConfig with test data
+	appConfig = mockEconomicConfig()
+
+	// Create a test user with a unique api_key
+	user := &models.User{
+		Username:       "testuser",
+		AccountBalance: 1000,
+		ApiKey:         "unique_api_key_1", // Ensure this is unique
+	}
+	db.Create(&user) // Save the user to the database
+
+	// Scenario 1: User places a bet on Market 1 where they have no prior bets
+	initialBetFee := getUserInitialBetFee(db, 1, user)
+	if initialBetFee != appConfig.Economics.Betting.BetFees.InitialBetFee {
+		t.Errorf("Expected initial bet fee to be %d, got %d", appConfig.Economics.Betting.BetFees.InitialBetFee, initialBetFee)
+	}
+
+	// Place a bet for the user on Market 1
+	bets := []models.Bet{
+		{Username: "testuser", MarketID: 1, Amount: 100, PlacedAt: time.Now()},
+	}
+	db.Create(&bets) // Save the bet to the database
+
+	// Scenario 2: User places another bet on Market 1 where they already have a bet
+	initialBetFee = getUserInitialBetFee(db, 1, user)
+	if initialBetFee != 0 {
+		t.Errorf("Expected initial bet fee to be 0, got %d", initialBetFee)
+	}
+
+	// Scenario 3: User places a bet on Market 2 where they have no prior bets
+	initialBetFee = getUserInitialBetFee(db, 2, user)
+	if initialBetFee != appConfig.Economics.Betting.BetFees.InitialBetFee {
+		t.Errorf("Expected initial bet fee to be %d, got %d", appConfig.Economics.Betting.BetFees.InitialBetFee, initialBetFee)
+	}
+
+}
+
+func TestGetTransactionFee(t *testing.T) {
+	// Mock the appConfig with test data
+	appConfig = mockEconomicConfig()
+
+	// Test buy scenario
+	buyBet := models.Bet{Amount: 100}
+	transactionFee := getTransactionFee(buyBet)
+	if transactionFee != appConfig.Economics.Betting.BetFees.BuySharesFee {
+		t.Errorf("Expected buy transaction fee to be %d, got %d", appConfig.Economics.Betting.BetFees.BuySharesFee, transactionFee)
+	}
+
+	// Test sell scenario
+	sellBet := models.Bet{Amount: -100}
+	transactionFee = getTransactionFee(sellBet)
+	if transactionFee != appConfig.Economics.Betting.BetFees.SellSharesFee {
+		t.Errorf("Expected sell transaction fee to be %d, got %d", appConfig.Economics.Betting.BetFees.SellSharesFee, transactionFee)
+	}
+}
+
+func TestGetSumBetFees(t *testing.T) {
+	// Set up in-memory SQLite database
+	db, err := gorm.Open(sqlite.Open(":memory:"), &gorm.Config{})
+	if err != nil {
+		t.Fatalf("Failed to connect to the database: %v", err)
+	}
```

Comment:
I think this is also a good candidate for moving into a `modelstesting.go` file that could be used to export in testing. We could make the function something like `func MustMakeFakeDB(t *testing.T) *gorm.DB` that way if it fails to return the fake, the test will fail without each test using the fake needing to check for connection to the fake.

Reply:
Created ticket https://github.com/orgs/openpredictionmarkets/projects/6/views/1?pane=issue&itemId=78968548